charmcraft/commands/store/store.py

```diff
@@ -35,24 +35,193 @@
+    """
+
+    message: str
+    code: str
+
+
+@dataclasses.dataclass(frozen=True)
+class Uploaded:
+    """Charmcraft-specific store upload result model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    ok: bool
+    status: int
+    revision: int
+    errors: List[Error]
+
+
 # XXX Facundo 2020-07-23: Need to do a massive rename to call `revno` to the "revision as
 # the number" inside the "revision as the structure", this gets super confusing in the code with
 # time, and now it's the moment to do it (also in Release below!)
-Revision = namedtuple("Revision", "revision version created_at status errors bases")
-Error = namedtuple("Error", "message code")
-Release = namedtuple("Release", "revision channel expires_at resources base")
-Channel = namedtuple("Channel", "name fallback track risk branch")
-Library = namedtuple("Library", "api content content_hash lib_id lib_name charm_name patch")
-Resource = namedtuple("Resource", "name optional revision resource_type")
-ResourceRevision = namedtuple("ResourceRevision", "revision created_at size")
-RegistryCredentials = namedtuple("RegistryCredentials", "image_name username password")
-Base = namedtuple("Base", "architecture channel name")
+@dataclasses.dataclass(frozen=True)
+class Base:
+    """Charmcraft-specific store object base model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    architecture: str
+    channel: str
+    name: str
+
+
+@dataclasses.dataclass(frozen=True)
+class Revision:
+    """Charmcraft-specific store name revision model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    revision: int
+    version: Optional
+    created_at: datetime.datetime
+    status: str
+    errors: List[Error]
+    bases: List[Base]
+
+
+@dataclasses.dataclass(frozen=True)
+class Resource:
+    """Charmcraft-specific store name resource model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    name: str
+    optional: bool
+    revision: int
+    resource_type: str
+
+
+@dataclasses.dataclass(frozen=True)
+class ResourceRevision:
+    """Charmcraft-specific store resource revision model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    revision: int
+    created_at: datetime.datetime
+    size: int
+
+
+@dataclasses.dataclass(frozen=True)
+class Release:
+    """Charmcraft-specific store release model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    revision: int
+    channel: str
+    expires_at: datetime.datetime
+    resources: List[Resource]
+    base: Base
+
+
+@dataclasses.dataclass(frozen=True)
+class Channel:
+    """Charmcraft-specific store channel model.
+
+    Deprecated in favour of implementation in craft-store.
+    """
+
+    name: str
+    fallback: str
+    track: str
+    risk: str
+    branch: str
+
+
+@dataclasses.dataclass(frozen=True)
+class Library:
```

Comment:
I guess the order doesn't matter?  This one is the only one I noticed that is in a different order.

Reply:
No, these are just internal data structures